CONTRIBUTING.md

```diff
@@ -0,0 +1,137 @@
+# Contributing to Consul.NET
+
+We'd love for you to contribute to our source code and to make Consul.NET even better than it is
+today! Here are the guidelines we'd like you to follow:
+
+ - [Code of Conduct](CODE_OF_CONDUCT.md)
+ - [Question or Problem?](#question)
+ - [Issues and Bugs](#issue)
+ - [Feature Requests](#feature)
+ - [Submission Guidelines](#submit)
+ - [Coding Rules](#rules)
+
+## <a name="issue"></a> Found an Issue?
+
+If you find a bug in the source code or a mistake in the documentation, you can help us by
+submitting an issue to our [GitHub](https://github.com/G-Research/consuldotnet/issues) repository. Even better you can submit a Pull Request with a fix.
+
+**Please see the [Submission Guidelines](#submit) below.**
+
+## <a name="feature"></a> Want a Feature?
+
+You can request a new feature by submitting an issue to our [GitHub](https://github.com/G-Research/consuldotnet/issues) repository. If you
+would like to implement a new feature then consider what kind of change it is:
+
+* **Major Changes** that you wish to contribute to the project should be discussed first in an issue, so that we can better coordinate our efforts,
+  prevent duplication of work, and help you to craft the change so that it is successfully accepted
+  into the project.
+* **Small Changes** can be crafted and submitted to the [GitHub](https://github.com/G-Research/consuldotnet/pulls) Repository as a Pull
+  Request.
+
+## <a name="submit"></a> Submission Guidelines
+
+### Submitting an Issue
+
+If your issue appears to be a bug, and hasn't been reported, open a new issue. Help us to maximize the effort we can spend fixing issues and adding new features, by not reporting duplicate issues.
+
+Providing the following information will increase the chances of your issue being dealt with
+quickly:
+
+* **Overview of the Issue** - if an error is being thrown a stack trace helps
+* **Motivation for or Use Case** - explain why this is a bug for you
+* **Consul/Consul.NET Version(s)** - is it a regression?
+* **Operating System** - is this a problem with all platforms or only specific ones?
+* **Reproduce the Error** - provide a example or an unambiguous set of steps.
+* **Related Issues** - has a similar issue been reported before?
+* **Suggest a Fix** - if you can't fix the bug yourself, perhaps you can point to what might be causing the problem (line of code or commit)
+
+**If you get help, help others. Good karma rulez!**
+
+### Submitting a Pull Request
+Before you submit your pull request consider the following guidelines:
+
+* Search [GitHub](https://github.com/G-Research/consuldotnet) for an open or closed Pull Request that relates to your submission. You don't want to duplicate effort.
+* Make your changes in a new git branch:
+
+    ```shell
+    git checkout -b my-fix-branch develop
+    ```
+
+* Create your patch, **including appropriate test cases**.
+* Follow our [Coding Rules](#rules).
+* Run the test suite, as described below.
```

Comment:
There doesn't seem to be a description of how to run tests below, maybe "Run the test suite using the development environment described below"?

Reply:
Done